tools/diagnostics/nsdiff

```diff
@@ -371,29 +465,99 @@ class NodeStateComparator():
+                            logger.debug(debug_message.format(filename))
+                            leveldb_rocksdb_handle = initKeyValueStorage(
+                                KeyValueStorageType.Rocksdb, dirname, filename,
+                                read_only=True)
+                            opened = True
+                        except:
+                            # rocksdb.errors.InvalidArgument
+                            e = sys.exc_info()[0]
+                            debug_message = ("Failed opening {} as a"
+                                             "RocksDB instance using"
+                                             "initKeyValueStorage with"
+                                             "exception: {}")
+                            logger.debug(debug_message.format(filename, e))
+                            try:
+                                debug_message = ("Retry opening {} as a RocksDB"
+                                                 "instance with"
+                                                 "initKeyValueStorageIntKeys...")
+                                logger.debug(debug_message.format(filename))
+                                leveldb_rocksdb_handle = initKeyValueStorageIntKeys(
+                                    KeyValueStorageType.Rocksdb, dirname,
+                                    filename, read_only=True)
+                                opened = True
+                            except:
+                                e = sys.exc_info()[0]
+                                debug_message = ("Failed opening {} as a"
+                                                 "RocksDB instance using"
+                                                 "initKeyValueStorageIntKeys"
+                                                 "with exeption {}")
+                                logger.debug(debug_message.format(filename, e))
+                                error_message = ("Failed to open {}/{} as a"
+                                                 "LevelDB or RocksDB instance")
+                                logger.error(error_message.format(dirname,
+                                    filename, e))
+                                opened = False
+
+
+                    if opened:
+                        debug_msg = ("Successfully opened LevelDB/RocksDB"
+                                     "instance {}")
+                        logger.debug(debug_msg.format(filename))
                         out_filename = join(sout, filename)
                         with open(out_filename, "a") as file_handle:
-                            if (filename == "state_signature" and not
-                                self.replay):
-                                for k, v in leveldb_handle.iterator():
+                            if (filename == "state_signature"):
+                                for k, v in leveldb_rocksdb_handle.iterator():
```

Comment:
You can break this into functions like `digest_state_signature`, `digest_state`, `digest_merkleLeaves`, etc